doc/wallet-passphrase.md

```diff
@@ -0,0 +1,21 @@
+## Wallet Passphrase
+
+Understanding the nuances of wallet security is essential to storing your Bitcoin safely. One crucial aspect of safeguarding funds is the utilization of a wallet passphrase or password, implemented through wallet encryption introduced in Bitcoin Core [v0.4.0](https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-0.4.0.md). Let's explore key points to demystify the nature of a wallet passphrase and the encryption process, emphasizing what they do and what they don't do:
```

Comment:
No need to say "introduced in" especially since v0.4.0 was not Bitcoin Core in any meaningful way